resources/sdk/clisdkclient/scripts/post-process.js

```diff
@@ -47,28 +47,53 @@ var {{=it.supercommand}}Cmd = &cobra.Command{
 func Cmd{{=it.supercommand}}() *cobra.Command {
 	return {{=it.supercommand}}Cmd
 }
-`
+`;
 	for (const supercommand of topLevelCommands) {
-		const commandPath = path.join(rootDir, supercommand)
-		const commandFile = path.join(commandPath, `${supercommand}.go`)
+		const commandPath = path.join(rootDir, supercommand);
+		const commandFile = path.join(commandPath, `${supercommand}.go`);
 		if (!fs.existsSync(commandFile)) {
-			fs.mkdirSync(commandPath)
-			writeTemplate(templateString, { supercommand: supercommand }, commandFile)
+			fs.mkdirSync(commandPath);
+			writeTemplate(templateString, { supercommand: supercommand }, commandFile);
 		}
 	}
 }
 
 // Generates root files to attach subcommands to supercommands
-function generateRootFiles(rootDir, resourceDefinitions, duplicateMappings) {
-	let commandMappings = new Map()
+function generateRootFiles(rootDir, resourceDefinitions, duplicateMappings, topLevelCommands) {
+	let commandMappings = new Map();
 	for (const path of Object.keys(resourceDefinitions)) {
-		const supercommand = resourceDefinitions[path].supercommand
-		if (supercommand !== undefined) {
-			let entry = commandMappings.get(supercommand) || new Set()
-			const key = `${resourceDefinitions[path].name}_${supercommand}`
-			const value = duplicateMappings[key]
-			entry.add(value ? key : resourceDefinitions[path].name)
-			commandMappings.set(supercommand, entry)
+		var commandName = resourceDefinitions[path].name;
+		if (resourceDefinitions[path].supercommand) {
+			const supercommandlist = resourceDefinitions[path].supercommand.split('.');
+
+			let nonRootSuperCommands = new Set();
+			for (const command of supercommandlist) {
+				if (!topLevelCommands.includes(command)) {
+					nonRootSuperCommands.add(command);
+				}
+			}
+
+			generateSuperCommandFiles(rootDir, nonRootSuperCommands);
+
+			supercommandlist.push(commandName);
+
+			for (var i = supercommandlist.length - 1; i >= 1; i--) {
```

Comment:
When possible use a foreach loop.  Using length and booleans are susceptible to post errors